modules/internal/BytesStringCommon.chpl

```diff
@@ -1072,6 +1072,43 @@ module BytesStringCommon {
     }
   }
 
+  proc doAppendSomeBytes(ref lhs: ?t, n: int, byteCArr: c_array)
```

Comment:
Does `byteCArr: c_array(uint(8), ?))` not work?

Reply:
Well... I think `n` can technically be param and that we can have a fully instantiated type for `byteCArr`, too. But you didn't do that to avoid blowing up compilation time?

Reply:
I didn't do that because it can be convenient to work with a `c_array` allocated for a maximum size even though some might not be used. E.g. a character in UTF-8 can be up to 4 bytes, but it is often fewer.